### PR TITLE
tests: new assert()

### PR DESCRIPTION
**Pull request by edsantiago**
This is much scarier than I had intended. The intention is to
start migrating from 'expect_output' and to use 'assert'
instead; the reason is that 'assert' allows much more
flexibility, particularly negative tests.

We've long wanted something like "assert that output does
not contain 'foo'". I've been too lazy to implement it,
but last week I noticed code in bud.bats that does:

    ! expect_output "sdfsdfsdf"

This is a really super bad idea: although it works fine when
things are good and tests pass, it's a disaster when tests
fail because the poor person debugging test output now sees
a test failure log, and starts tracking it down, and then
much later realizes that it was an intentional failure but
the '!' was negating it, and all that track-down work was
wasted.

To keep this PR reasonable, I'm keeping all positive uses
of expect_output untouched. We can migrate those (if desired)
over time. All negative uses are now assertions.

Signed-off-by: Ed Santiago <santiago@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
@edsantiago still working on this?

**Comment by edsantiago** [comment]
I still lean slightly toward thinking this is a good idea, but I don't want to pursue it unilaterally. Could I get a second opinion? Would this make the tests easier to write/maintain?

**Comment by rhatdan** [comment]
Yes I think this is a good idea.  I think it reads easier and handling of string should not exists, is difficult.

**Comment by edsantiago** [comment]
Thanks. I've put this back on my shortlist.

**Comment by edsantiago** [comment]
@containers/buildah-maintainers opinions welcome

**Comment by rhatdan** [comment]
LGTM
@nalind @TomSweeneyRedHat @vrothberg @umohnani8 @saschagrunert PTAL
@containers/podman-maintainers PTAL

**Review by vrothberg — APPROVED** [review]
/lgtm

**Pull request merged**
